pkg/runner/action.go

```diff
@@ -371,11 +371,14 @@ func newStepContainer(ctx context.Context, step step, image string, cmd []string
 }
 
 func populateEnvsFromSavedState(env *map[string]string, step actionStep, rc *RunContext) {
-	stepResult := rc.StepResults[step.getStepModel().ID]
-	if stepResult != nil {
-		for name, value := range stepResult.State {
-			envName := fmt.Sprintf("STATE_%s", name)
-			(*env)[envName] = value
+	// design problem: you can create steps with an id prefixed with pre-
```

Comment:
I think I don't get this. Do we have a test case for that?
And why is it a problem if there is a step with a `pre-` id?

Reply:
Ouch, if that's the case it would be a critical error. Right.

Reply:
But the `step.ID` is not changed for 'pre-steps'. It's just the `step.Name` which is prefixed.
The ID should be stable and in combination with the step stage it indicates the stage of that step.

Reply:
I see, we use the `CurrentStep` in the command handler.
Wouldn't it be possible to add the `CurrentStepID` to the `RunContext` as well. That should be way more stable than a name.

Reply:
> Shouldn't we remove presteps from the steps context, called stepresults? To align to actions/runner.

Would be a good idea I think.